.upstream-tests/test/support/cuda_space_selector.h

```diff
@@ -9,22 +9,26 @@
 
 #include <cuda/std/cstddef>
 
-#if defined(__CUDACC__) && !defined(__NVCOMPILER)
-#define __exec_check_disable__ #pragma nv_exec_check_disable
+#if defined(__CUDACC__) && (defined(__CUDACC_RTC__) || defined(__NVCC__))
+# define __exec_check_disable__ #pragma nv_exec_check_disable
 #else
-#define __exec_check_disable__
+# define __exec_check_disable__
 #endif
 
 #ifdef __CUDACC_RTC__
-#define LAMBDA [=]
+# define LAMBDA [=]
 #else
-#define LAMBDA [=] __host__ __device__
+# define LAMBDA [=] __host__ __device__
 #endif
 
 #ifdef __CUDA_ARCH__
-#define SHARED __shared__
+# define SHARED __shared__
 #else
-#define SHARED
+# define SHARED
+#endif
+
+#if defined(__clang__) && defined(__CUDA__)
```

Comment:
e.g., https://github.com/NVIDIA/libcudacxx/pull/304/files#r950113474

Reply:
We generally try to refrain from using libcudacxx macros in the tests. This allows us to ensure the behavior matches up.